modules/buildkite-agent.nix

```diff
@@ -14,8 +14,16 @@
     openssh.publicKeyPath  = "/run/keys/buildkite-ssh-public";
     tokenPath              = "/run/keys/buildkite-token";
     meta-data              = "system=x86_64-linux";
-    hooksPath              = "/var/lib/buildkite-agent/hooks";
     runtimePackages        = with pkgs; [ gnutar gzip bzip2 xz ];
+    hooks.pre-command = ''
+      NIX_PROFILE_BINPATH=/run/current-system/sw/bin
+      NIX_BUILD="$NIX_PROFILE_BINPATH/nix-build"
+      NIX_PATH="nixpkgs=$(''${NIX_BUILD} fetch-nixpkgs.nix -o nixpkgs)"
+      export NIX_REMOTE=daemon
+      export PATH="$NIX_PROFILE_BINPATH:$PATH"
+      export NIX_PATH
+      source /var/lib/buildkite-agent/hooks/pre-command
```

Comment:
This seems unlikely to continue to work because it relies on state that's not set up by nixos configuration.

Reply:
Oh hang on, that file is set up by `deployment.keys`. Maybe change the filename.